bin/network-monitor/src/main.rs

```diff
@@ -34,44 +40,71 @@ async fn main() -> anyhow::Result<()> {
         },
     };
 
-    // Initialize shared status state
-    let shared_status: SharedStatus =
-        Arc::new(Mutex::new(NetworkStatus { services: Vec::new(), last_updated: 0 }));
-
-    // Start tasks for frontend and status monitoring
-    let mut join_set = JoinSet::new();
-    let mut component_ids = HashMap::new();
-
-    // Spawn frontend task
-    let frontend_status = shared_status.clone();
-    let frontend_config = config.clone();
-    let id = join_set
-        .spawn(async move { serve(frontend_status, frontend_config).await })
-        .id();
-    component_ids.insert(id, "frontend");
-
-    // Spawn status monitoring task
-    let status_status = shared_status.clone();
-    let status_config = config.clone();
-    let id = join_set
-        .spawn(async move { check_status(status_status, status_config).await })
-        .id();
-    component_ids.insert(id, "status");
+    let mut tasks = JoinSet::new();
+    let mut component_ids: HashMap<_, String> = HashMap::new();
+
+    // Create initial empty status for services
+    let initial_rpc_status = ServiceStatus {
+        name: "RPC".to_string(),
+        status: "unknown".to_string(),
+        last_checked: 0,
+        error: None,
+        details: None,
+    };
+
+    // Spawn the RPC checker
+    let (rpc_tx, rpc_rx) = watch::channel(initial_rpc_status);
+    let rpc_url = config.rpc_url.clone();
+    let id = tasks.spawn(async move { run_rpc_status_task(rpc_url, rpc_tx).await }).id();
+    component_ids.insert(id, "rpc-checker".to_string());
+
+    // Spawn the prover checkers
+    let mut prover_rxs = Vec::new();
+    for (i, prover_url) in config.remote_prover_urls.iter().enumerate() {
+        let name = format!("Prover-{}", i + 1);
+        let initial_prover_status = ServiceStatus {
+            name: format!("Remote Prover ({name})"),
+            status: "unknown".to_string(),
+            last_checked: 0,
+            error: None,
+            details: None,
+        };
+
+        let (prover_tx, prover_rx) = watch::channel(initial_prover_status);
+        prover_rxs.push(prover_rx);
+
+        let prover_url_clone = prover_url.clone();
+        let name_clone = name.clone();
+        let component_name = format!("prover-checker-{}", i + 1);
+        let id = tasks
+            .spawn(async move {
+                run_remote_prover_status_task(prover_url_clone, name_clone, prover_tx).await
+            })
+            .id();
```

Comment:
Nit: you can move the clones "inside" to prevent the naming pains


```suggestion
        let component_name = format!("prover-checker-{}", i + 1);
        let id = tasks.spawn({
                let prover_url = prover_url.clone();
                let name = name.clone();
                
                run_remote_prover_status_task(prover_url, name, prover_tx)
            })
            .id();
```